### ISSUE TITLE
Ability to make this module a replica, then promote it to a master

### ISSUE THREAD

**Issue by OscarVanL** [terraform-google-modules/terraform-google-sql-db#554]
### TL;DR

I am trying to follow the Google docs [here](https://cloud.google.com/sql/docs/mysql/replication/configure-replication-from-external#setup-replica-instance) to create a replica instance from my on-premises database.

My intent is to use this module as a means to migrate from an on-premises MySQL database to CloudSQL via replication.

The "root" database this module creates will start as a replica, then later get promoted to master.

### Terraform Resources

First I want to set my [google_sql_source_representation_instance](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/sql_source_representation_instance) (representing my on-premises DB) as the `master_instance_name` and set the `instance_type` to `READ_REPLICA_INSTANCE` for the "root" [google_sql_database_instance](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/sql_database_instance) that this module creates.

Then later I want to promote this instance to the master by unsetting `master_instance_name` and setting `instance_type` to `CLOUD_SQL_INSTANCE`.

Some more details about the exact process for promoting a replica instance can be seen here: https://github.com/GoogleCloudPlatform/magic-modules/pull/7138

[google_sql_source_representation_instance](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/sql_source_representation_instance)

[google_sql_database_instance](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/sql_database_instance)

### Detailed design

It looks like what this module implements is exactly what I need, except the ability to specify `master_instance_name` and `instance_type` on the "root" database.

**Comment by OscarVanL** [comment]
An update from my side. Using the changes I made here I successfully performed a live migration from my self-hosted databases to CloudSQL within my company.

1. Create a `google_sql_source_representation_instance` referencing your self-hosted instance you're migrating from.
2. Use this module to create a CloudSQL instance with the name of that source representation instance as its `master_instance_name`. This will setup a replica. (It is not actually necessary to set `instance_type` to `READ_REPLICA_INSTANCE` as I originally thought in the issue description).
3. Initiate replication using the relevant API calls. This can't be terraformed as far as I am aware. I opted for the [managed import](https://cloud.google.com/sql/docs/mysql/replication/managed-import-replication-from-external) approach.
4. Create all the necessary replicas, config, etc on your CloudSQL instance via this module.
5. When you're ready, promote the CloudSQL replica to the master by unassigning `master_instance_name` and setting `instance_type` to `CLOUD_SQL_INSTANCE`.

This approach allows you to do a migration from a self-hosted database into CloudSQL, all while using this module, and avoiding any tricky steps where you must create the resources individually and have to import them later.
